From b1707944eacbb34b0704f8bd8362c19dfb3d9047 Mon Sep 17 00:00:00 2001
Message-Id: <b1707944eacbb34b0704f8bd8362c19dfb3d9047.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Sangani Suryanarayana Raju <suryanarayana.r.sangani@intel.com>
Date: Sat, 8 Nov 2014 18:43:43 +0530
Subject: [PATCH 20/34] SQUASHME! [VPG]: drm/i915/vlv: Fix default value of
 scaling mode property

SQUASHME! -  This patch should be squashed into the following existing patch:
	Author: Vidya Srinivas <vidya.srinivas@intel.com>
	Change-Id: Ie5a93f5ad4f26bfa00e5eed329e0146bea799a1f
	FOR_UPSTREAM [VPG]: drm/i915: Panel fitter/Scaling Property added

For VLV and CHT, we don't want to use DRM_MODE_SCALE_ASPECT. Using this
causes eDP blank out during boot up.

Issue: GMINL-3538
Change-Id: I64ca7cf7a14ab99b10e3166ec64338a859ebcaea
Signed-off-by: Sangani Suryanarayana Raju <suryanarayana.r.sangani@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index d430e4b..438e272 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -4920,6 +4920,7 @@ static void
 intel_dp_add_properties(struct intel_dp *intel_dp, struct drm_connector *connector)
 {
 	struct intel_connector *intel_connector = to_intel_connector(connector);
+	struct drm_device *dev = connector->dev;
 
 	intel_attach_force_audio_property(connector);
 	intel_attach_broadcast_rgb_property(connector);
@@ -4927,7 +4928,7 @@ intel_dp_add_properties(struct intel_dp *intel_dp, struct drm_connector *connect
 	intel_attach_scaling_src_size_property(connector);
 	intel_dp->color_range_auto = true;
 
-	if (is_edp(intel_dp)) {
+	if (is_edp(intel_dp) && !IS_VALLEYVIEW(dev)) {
 		drm_mode_create_scaling_mode_property(connector->dev);
 		drm_object_attach_property(
 			&connector->base,
-- 
1.7.9.5

